src/components/Form/Form.js

```diff
@@ -46,7 +46,11 @@ export function Form({ children, config }) {
 
   // Set up initial values
   let initialValues = {}
-  fieldNames.forEach((x) => {
+  fieldNames.forEach((fieldName) => {
+    // We don't do the preinitialize to non empty string trick for optional
+    // fields as we'd like those to only be validated upon entering something
+    if (!!config.fields[fieldName].optional) return
```

Comment:
This is essentially what makes it all work really. @kenning just kinda going along with your thing here as a means to consider a non interacted with field valid. Seems to work great ;)